data/database/migration_rollback.py

```
@@ -0,0 +1,610 @@
+            
+            self.logger.info(f"Created database backup: {backup_path}")
+            return str(backup_path)
+            
+        except Exception as e:
+            error_msg = f"Failed to create database backup: {e}"
+            self.logger.error(error_msg)
+            raise BaseMigrationError(
+                message=error_msg,
+                context=None,
+                original_error=e
+            )
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)